orange/src/test/java/com/hartwig/hmftools/orange/algo/linx/LinxBreakendInterpreterTest.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    @Test
+    public void canFallbackWhenNoStructuralVariant()
+    {
+        List<StructuralVariant> structuralVariants = List.of();
+
+        List<LinxSvAnnotation> linxSvAnnotations = List.of(createSvAnnotation(1.0, 2.0));
+        List<LinxBreakend> breakends = createBreakends();
+
+        LinxBreakendInterpreter interpreter = createInterpreter(structuralVariants, linxSvAnnotations);
+        com.hartwig.hmftools.datamodel.linx.LinxBreakend left = interpreter.interpret(breakends.get(0));
+        assertEquals("", left.chromosome());
+        assertEquals("q34", left.chromosomeBand());
+        assertEquals(LinxBreakendType.BND, left.type());
+        assertEquals(0, left.orientation());
+        assertEquals(1.5D, left.junctionCopyNumber(), 0.0D);
+
+        com.hartwig.hmftools.datamodel.linx.LinxBreakend right = interpreter.interpret(breakends.get(1));
+        assertEquals("", right.chromosome());
+        assertEquals("q34", right.chromosomeBand());
+        assertEquals(LinxBreakendType.BND, right.type());
+        assertEquals(0, right.orientation());
+        assertEquals(1.5D, right.junctionCopyNumber(), 0.0D);
+    }
+    
+    @Test
+    public void canComputeJunctionCopyNumber()
+    {
+        LinxBreakend linxBreakend = createBreakend(1, 1, true, TranscriptRegionType.INTRONIC);
+        StructuralVariant sv = createStructuralVariant(StructuralVariantType.DUP);
+        LinxSvAnnotation svAnnotation = createSvAnnotation(1.0, 2.0);
+
+        double result = LinxBreakendInterpreter.junctionCopyNumber(svAnnotation);
+        assertEquals(1.5, result, EPSILON);
+    }
+
+    @NotNull
+    private StructuralVariant createStructuralVariant(@NotNull StructuralVariantType type)
+    {
+        return PurpleTestUtils.createStructuralVariant("7", 50, "7", 200, type, 1.0, 1.0)
+                .id(VCF_ID)
+                .build();
+    }
+
+    @NotNull
+    private List<LinxBreakend> createBreakends()
+    {
+        return List.of(
+                createBreakend(1, 1, true, TranscriptRegionType.INTRONIC),
+                createBreakend(2, 1, false, TranscriptRegionType.INTRONIC)
+        );
+    }
+
+    @NotNull
+    private LinxBreakend createBreakend(int id, int svId, boolean isStart, @NotNull TranscriptRegionType regionType)
+    {
+        return LinxTestFactory.breakendBuilder()
+                .id(id)
+                .reportedDisruption(true)
+                .gene(GENE)
+                .isStart(isStart)
+                .svId(svId)
+                .regionType(regionType)
+                .build();
+    }
+
+    @NotNull
+    private static LinxSvAnnotation createSvAnnotation(double junctionCopyNumberMin, double junctionCopyNumberMax)
+    {
+        return LinxTestFactory.svAnnotationBuilder()
+                .svId(1)
+                .vcfId(VCF_ID)
+                .junctionCopyNumberMin(junctionCopyNumberMin)
+                .junctionCopyNumberMax(junctionCopyNumberMax)
+                .build();
+    }
```

Comment:
For some of the above helper functions IDEA is warning that the variables always have the same value. You could consider varying these values a bit more between tests to show that they don't matter, or maybe just remove the option to set them in the test and hardcode them to something sensible in the helper function. I would be fine with changing nothing as well.

Reply:
Removed constant arguments and set values in the helpers.